packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -0,0 +1,365 @@
+    this.side,
+    this.shape,
+    this.padding,
+    this.textStyle,
+    this.hintStyle,
+  }) : assert(trailing == null || trailing.length <= 2);
+
+  /// Controls the text being edited.
+  ///
+  /// If null, this widget will create its own [TextEditingController].
+  final TextEditingController? controller;
+
+  /// {@macro flutter.widgets.Focus.focusNode}
+  final FocusNode? focusNode;
+
+  /// Text that suggests what sort of input the field accepts.
+  ///
+  /// Displayed at the same location on the screen where text may be entered
+  /// when the input is empty.
+  ///
+  /// Defaults to null.
+  final String? hintText;
+
+  /// A widget to display before the text input field.
+  ///
+  /// Typically the [leading] widget is an [Icon] or an [IconButton].
+  final Widget? leading;
+
+  /// A list of Widgets to display in a row after the text field.
+  ///
+  /// Typically these actions can represent additional modes of searching
+  /// (like voice search), an avatar, a separate high-level action (such as
+  /// current location) or an overflow menu. There should not be more than
+  /// two trailing actions.
+  final Iterable<Widget>? trailing;
+
+  /// Called when the user taps this search bar.
+  final GestureTapCallback? onTap;
+
+  /// Invoked upon user input.
+  final ValueChanged<String>? onChanged;
+
+  /// Optional size constraints for the search bar.
+  ///
+  /// If null, the value of [SearchBarThemeData.constraints] will be used. If
+  /// this is also null, then the constraints defaults to:
+  /// ```dart
+  /// const BoxConstraints(minWidth: 360.0, maxWidth: 800.0, minHeight: 56.0)
+  /// ```
+  final BoxConstraints? constraints;
+
+  /// The elevation of the search bar's [Material].
+  ///
+  /// If null, the value of [SearchBarThemeData.elevation] will be used. If this
+  /// is also null, then default value is 8.0.
+  final MaterialStateProperty<double?>? elevation;
+
+  /// The search bar's background fill color.
+  ///
+  /// If null, the value of [SearchBarThemeData.backgroundColor] will be used.
+  /// If this is also null, then the default value is [ColorScheme.surface].
+  final MaterialStateProperty<Color?>? backgroundColor;
+
+  /// The shadow color of the search bar's [Material].
+  ///
+  /// If null, the value of [SearchBarThemeData.shadowColor] will be used.
+  /// If this is also null, then the default value is [ColorScheme.shadow].
+  final MaterialStateProperty<Color?>? shadowColor;
+
+  /// The surface tint color of the button's [Material].
+  ///
+  /// See [Material.surfaceTintColor] for more details.
+  ///
+  /// If null, the value of [SearchBarThemeData.surfaceTintColor] will be used.
+  /// If this is also null, then the default value is [ColorScheme.surfaceTint].
+  final MaterialStateProperty<Color?>? surfaceTintColor;
+
+  /// The highlight color that's typically used to indicate that
+  /// the button is focused, hovered, or pressed.
+  final MaterialStateProperty<Color?>? overlayColor;
```

Comment:
These seem unecessary
- surface tint is going away (https://github.com/flutter/flutter/issues/115912)

Reply:
I see. And I noticed in the latest tokens, we removed all the surface tint color!😲